### PR TITLE
Look up trusted certs consistently on windows

### PR DESCRIPTION
**Pull request by amcasey**
1. Don't use thumbprint so we don't get flagged for using SHA-1
2. Make TrustCertificateCore and RemoveCertificateFromTrustedRoots consistent

### CODE REVIEW

**Comment by amcasey** [comment]
The CI failure is relevant.  I'll investigate.

File: src/Shared/CertificateGeneration/CertificateManager.cs [hunk 1]
```diff
@@ -812,6 +812,46 @@ internal static string ToCertificateDescription(IEnumerable<X509Certificate2> ce
...
+    /// <remarks>
+    /// <see cref="X509Certificate.Equals(X509Certificate?)"/> is not adequate for security purposes.
+    /// </remarks>
+    internal static bool AreCertificatesEqual(X509Certificate2 cert1, X509Certificate2 cert2)
+    {
+        return cert1.RawDataMemory.Span.SequenceEqual(cert2.RawDataMemory.Span);
+    }
+
+    /// <summary>
+    /// Given a certificate, usally from the <see cref="StoreName.My"/> store, try to find the
```
Comment (martincostello) [inline src/Shared/CertificateGeneration/CertificateManager.cs:824]: Nit:

```suggestion
    /// Given a certificate, usually from the <see cref="StoreName.My"/> store, try to find the
```

File: src/Shared/CertificateGeneration/CertificateManager.cs [hunk 2]
```diff
@@ -812,6 +812,46 @@ internal static string ToCertificateDescription(IEnumerable<X509Certificate2> ce
...
+                {
+                    DisposeCertificates(certificatesToDispose);
+                    foundCertificate = candidate;
+                    return true;
+                }
+                certificatesToDispose.Add(candidate);
+            }
+        }
+
+        foundCertificate = null;
```
Comment (adityamandaleeka) [inline src/Shared/CertificateGeneration/CertificateManager.cs:851]: Do we need to DisposeCertificates in the not found case too?

Comment (amcasey) [reply src/Shared/CertificateGeneration/CertificateManager.cs]: I'm not entirely sure why we have to call it in any case, but I agree that it should be consistent.

**Review by adityamandaleeka — APPROVED** [review]

**Pull request merged**